File: CS-Examples/02_FindAndReplace/ReplaceContentWithDoc.cs
using System;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;
using System.Text.RegularExpressions;

namespace ReplaceContentWithDoc
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new instance of the Document class
            Document document1 = new Document();
            // Load a Word document from a file
            document1.LoadFromFile(@"..\..\..\..\..\..\Data\ReplaceContentWithDoc.docx");
            // Create another instance of the Document class
            Document document2 = new Document();
            // Load another Word document from a file
            document2.LoadFromFile(@"..\..\..\..\..\..\Data\Insert.docx");
            // Get the first section of the first document
            Section section1 = document1.Sections[0];
            // Create a regular expression object to search for a pattern
            Regex regex = new Regex(@"\[MY_DOCUMENT\]", RegexOptions.None);
            // Find all occurrences of the pattern in the first document
            TextSelection[] textSections = document1.FindAllPattern(regex);
            // Loop through each occurrence of the pattern
            foreach (TextSelection seletion in textSections)
            {
                // Get the paragraph that contains the pattern
                Paragraph para = seletion.GetAsOneRange().OwnerParagraph;
                // Get the range of text that contains the pattern
                TextRange textRange = seletion.GetAsOneRange();
                // Get the index of the paragraph in the first document's section
                int index = section1.Body.ChildObjects.IndexOf(para);
                // Loop through each section in the second document
                foreach (Section section2 in document2.Sections)
                {
                    // Loop through each paragraph in the second section
                    foreach (Paragraph paragraph in section2.Paragraphs)
                    {
                        // Insert the paragraph from the second document into the first document's section
                        section1.Body.ChildObjects.Insert(index, paragraph.Clone() as Paragraph);
                    }
                }
                // Remove the range of text that contains the pattern from the paragraph
                para.ChildObjects.Remove(textRange);
            }
            // Save the modified first document to a file
            document1.SaveToFile("Output.docx", FileFormat.Docx);
            // Dispose the first document and release all resources it is using
            document1.Dispose();
            // Dispose the second document and release all resources it is using
            document2.Dispose();

            //Launch the Word file.
            WordDocViewer("Output.docx");
        }
        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB-Examples/02_FindAndReplace/ReplaceContentWithDoc.vb
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields
Imports System.Text.RegularExpressions

Namespace ReplaceContentWithDoc
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new Document object and load the first document from the specified file path
			Dim document1 As New Document()
			document1.LoadFromFile("..\..\..\..\..\..\Data\ReplaceContentWithDoc.docx")

			' Create a new Document object and load the second document from the specified file path
			Dim document2 As New Document()
			document2.LoadFromFile("..\..\..\..\..\..\Data\Insert.docx")

			' Get the first section of document1
			Dim section1 As Section = document1.Sections(0)

			' Create a regular expression pattern to match "[MY_DOCUMENT]"
			Dim regex As New Regex("\[MY_DOCUMENT\]", RegexOptions.None)

			' Find all text sections in document1 that match the pattern
			Dim textSections() As TextSelection = document1.FindAllPattern(regex)

			' Iterate through each TextSelection
			For Each selection As TextSelection In textSections
				' Get the owning Paragraph and TextRange of the selection
				Dim para As Paragraph = selection.GetAsOneRange().OwnerParagraph
				Dim textRange As TextRange = selection.GetAsOneRange()

				' Get the index of the owning Paragraph within section1
				Dim index As Integer = section1.Body.ChildObjects.IndexOf(para)

				' Iterate through each Section in document2
				For Each section2 As Section In document2.Sections
					' Iterate through each Paragraph in section2 and insert a clone into section1 at the specified index
					For Each paragraph As Paragraph In section2.Paragraphs
						section1.Body.ChildObjects.Insert(index, TryCast(paragraph.Clone(), Paragraph))
					Next paragraph
				Next section2

				' Remove the textRange from the owning Paragraph
				para.ChildObjects.Remove(textRange)
			Next selection

			' Save the modified document1 to a file named "Output.docx"
			document1.SaveToFile("Output.docx", FileFormat.Docx)

			' Dispose of the Document objects to release resources
			document1.Dispose()
			document2.Dispose()

			'Launch the Word file.
			WordDocViewer("Output.docx")
		End Sub
		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
